assets/js/type-defs/checkout.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @typedef {Object} CheckoutDispatchActions
+ *
+ * @property {function()} resetCheckout        Dispatches an action that resets
+ *                                           the checkout to a pristine state.
+ * @property {function()} setRedirectUrl       Dispatches an action that sets the
```

Comment:
why the mixed usage of `function()` and `Function` in prop types

Reply:
Good question!  Neither `Function` or `function()` are jsdoc spec but we had been using `Function` to denote variables that are a function in our code. However with the typescript linting, `function()` works better because eventually we (and I have in some places already) can do things like `function( string, number ):number` for better type-hinting. My example in this case would indicate a function that receives two arguments, one a string and one a number and returns a number.

I've just held off going full out on this enhanced documentation until I learn what works best. So places where you see `Function` is just something I haven't addressed yet for typescript linting. Places where you see `function()` has been the minimum I've needed to address Typescript linting for functions, and places where you see `function( something ):something` are places where I spent a bit extra time to see what works for accurate typescript linting.

I'm not going to be picky on reviews (and I don't think we should in general yet) until the cooldown when I can do some documentation and the squad can have some discussion about what conventions we want to adopt.

Reply:
leaving this unresolved for visibility.